lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateTaskAccTest.java

```diff
@@ -0,0 +1,582 @@
+
+    defaultObjectReference = defaultTestObjectReference().build();
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdatePrimaryObjectReferenceOfTask_When_Requested() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService);
+    task.setPrimaryObjRef(
+        createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", "VNR", "7654321"));
+
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
+    assertThat(updatedTask.getPrimaryObjRef().getCompany()).isEqualTo("COMPANY_A");
+    assertThat(updatedTask.getPrimaryObjRef().getSystem()).isEqualTo("SYSTEM_A");
+    assertThat(updatedTask.getPrimaryObjRef().getSystemInstance()).isEqualTo("INSTANCE_A");
+    assertThat(updatedTask.getPrimaryObjRef().getType()).isEqualTo("VNR");
+    assertThat(updatedTask.getPrimaryObjRef().getValue()).isEqualTo("7654321");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdateBusinessAndParentBusinessProcessIdOfTask_When_Requested() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService);
+    task.setBusinessProcessId("MY_PROCESS_ID");
+    task.setParentBusinessProcessId("MY_PARENT_PROCESS_ID");
+
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
+    assertThat(updatedTask.getBusinessProcessId()).isEqualTo("MY_PROCESS_ID");
+    assertThat(updatedTask.getParentBusinessProcessId()).isEqualTo("MY_PARENT_PROCESS_ID");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UpdateTimeStampsAndFlagsOfTask_When_Requested() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService);
+    final Instant modifiedOriginal = task.getModified();
+    task.setBusinessProcessId("MY_PROCESS_ID");
+
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
+    assertThat(updatedTask.getCreated()).isNotNull();
+    assertThat(updatedTask.getModified()).isNotNull();
+    assertThat(modifiedOriginal.isAfter(updatedTask.getModified())).isFalse();
+    assertThat(updatedTask.getModified()).isNotEqualTo(updatedTask.getCreated());
```

Comment:
This check could make the test flaky. If the test runs fast enough, the values of created and modified might be equal. @gitgoodjhe, what do you think about adding sleep after task creation in this test?

Reply:
In principle I agree. Are we running with nanosecond prescision or millisecond precision? For nano the question should be out of the window, but even for milli it is highly unlikely that the updateTask runs <1ms.

I think we have similar tests with timestamps that so far never failed